### PR TITLE
Support passing sandbox as argument and iteration reminder

### PR DESCRIPTION
**Pull request by xingyaoww**
This is part of the modification made in #1468. 

This PR:
- support passing sandbox as an argument, which is useful in SWEBench evaluation, where the `run_infer.py` script will pass in a pre-initialized `SWEBenchEnvBox`.
- supports `iteration_reminder` that appends a count-down message to the agent in SWE-Bench evaluation (default to false).

### CODE REVIEW

**Review by li-boxuan — APPROVED** [review]
Is it possible to set `iteration_reminder` to True in integration tests? ~~We don't have a way to verify this log in our tests right now, but~~ at least it will ensure the code is covered by tests (higher code coverage!)

**Comment by xingyaoww** [comment]
@li-boxuan Good idea... Let me figure out if there's a way to patch the value of `config.iteration_reminder` to true in the tests..

File: tests/integration/README.md [hunk 1]
```diff
@@ -20,11 +20,10 @@ not possible with benchmarks.
...
 Known limitations:
 1. To avoid the potential impact of non-determinism, we remove all special
-characters and numbers (often used as PIDs) when doing the comparison. If two
-prompts for the same task only differ in non-alpha characters, a wrong mock
-response might be picked up.
-2. It is required that the agent itself doesn't do anything non-deterministic,
-including but not limited to using randomly generated numbers.
+characters when doing the comparison. If two prompts for the same task only
+differ in non-alphanumeric characters, a wrong mock response might be picked up.
+2. It is required that everything has to be deternimistic. For example, agent
```
Comment (neubig) [inline tests/integration/README.md:25]:
```suggestion
2. It is required that everything has to be deterninistic. For example, agent
```

**Review by neubig — APPROVED** [review]
Looks great! We might eventually want to consider having remind_iterations on by default, but it doesn't have to be in this PR

File: tests/integration/README.md (hunk 1 above)
Comment (xingyaoww) [reply tests/integration/README.md:25]: Should it be `deterministic`? ;)

Comment (xingyaoww) [inline tests/integration/README.md:25]:
```suggestion
2. It is required that everything has to be deterministic. For example, agent
```

**Pull request merged**